packages/maps/src/index.ts

```diff
@@ -1,54 +1,147 @@
+
+const DEFAULT_SERVER_URL_TEMPLATE = 'https://{user}.carto.com';
+const DEFAULT_CLIENT_ID = 'maps-pkg';
+const REQUEST_GET_MAX_URL_LENGTH = 2048;
+
 class Maps {
-  private _username: string;
-  private _apiKey: string;
+  private _conf: any;
+  private _auth: string;
+  private _client: string;
 
-  constructor(username: string, apiKey: string) {
-    this._username = username;
-    this._apiKey = apiKey;
-  }
+  /**
+   * Build an instance to interact with Maps API v1 (aka Windshaft)
+   * @param username
+   * @param apiKey
+   * @param serverURL A url pattern like default ('https://{user}.carto.com')
+   *
+   */
+  constructor(username: string, apiKey: string, serverURL?: string, client?: string) {
 
-  public get username(): string {
-    return this._username;
+    this._conf = {
+      username,
+      apiKey,
+      serverUrlTemplate: serverURL || DEFAULT_SERVER_URL_TEMPLATE
+    };
+
+    this._auth = this.encodeParameter('api_key', this._conf.apiKey);
+    this._client = this.encodeParameter('client', client || DEFAULT_CLIENT_ID);
   }
 
-  public set username(value: string) {
-    this._username = value;
+  public get username(): string {
+    return this._conf.username;
   }
 
   public get apiKey(): string {
-    return this._apiKey;
+    return this._conf.apiKey;
   }
 
-  public set apiKey(value: string) {
-    this._apiKey = value;
+  public get serverURL(): string {
+    return this._conf.serverUrlTemplate.replace('{user}', this._conf.username);
   }
 
-  public instantiateMap(options: { sql?: string, dataset?: string }) {
+  /**
+   * Instantiate a map based on dataset name or sql query, returning a layergroup
+   *
+   * Note: Get layergroup.metadata.tilejson.vector.tiles to consume MVT tiles
+   * @param options
+   */
+  public async instantiateSimpleMap(options: { sql?: string, dataset?: string }) {
```

Comment:
Mmm, yep. I couldn't find a good name here. 

I'm thinking about something like a method 'overload', using the simple params { sql?: string, dataset?: string }, instead of the full mapConfig. Also, I wasn't sure about exposing now the one with full mapConfig. 

Any thought?

Reply:
How about `instantiateMapFrom` (simplified) vs `instantiateMap` with the full mapconfig (private, so far)?

Reply:
🔝 That's what I've finally done, rename old *instantiateSimpleMap* to **instantiateMapFrom**